cpp/src/randomforest/randomforest.cu

```diff
@@ -649,23 +604,28 @@ RF_metrics score(const raft::handle_t& user_handle,
   return classification_score;
 }
 
-RF_params set_rf_class_obj(int max_depth, int max_leaves, float max_features,
-                           int n_bins, int split_algo, int min_samples_leaf,
-                           int min_samples_split, float min_impurity_decrease,
-                           bool bootstrap_features, bool bootstrap, int n_trees,
-                           float max_samples, int seed,
-                           CRITERION split_criterion, bool quantile_per_tree,
-                           int cfg_n_streams, bool use_experimental_backend,
-                           int max_batch_size) {
+RF_params set_rf_params(int max_depth, int max_leaves, float max_features,
+                        int n_bins, int split_algo, int min_samples_leaf,
+                        int min_samples_split, float min_impurity_decrease,
+                        bool bootstrap_features, bool bootstrap, int n_trees,
+                        float max_samples, int seed, CRITERION split_criterion,
+                        bool quantile_per_tree, int cfg_n_streams,
+                        bool use_experimental_backend, int max_batch_size) {
   DecisionTree::DecisionTreeParams tree_params;
   DecisionTree::set_tree_params(
     tree_params, max_depth, max_leaves, max_features, n_bins, split_algo,
     min_samples_leaf, min_samples_split, min_impurity_decrease,
     bootstrap_features, split_criterion, quantile_per_tree,
     use_experimental_backend, max_batch_size);
   RF_params rf_params;
-  set_all_rf_params(rf_params, n_trees, bootstrap, max_samples, seed,
-                    cfg_n_streams, tree_params);
+  rf_params.n_trees = n_trees;
+  rf_params.bootstrap = bootstrap;
+  rf_params.max_samples = max_samples;
+  rf_params.seed = seed;
+  rf_params.n_streams = min(cfg_n_streams, omp_get_max_threads());
+  if (n_trees < rf_params.n_streams) rf_params.n_streams = n_trees;
+  DecisionTree::set_tree_params(rf_params.tree_params);
```

Comment:
Is this line actually necessary? The `rf_params.tree_params` field is going to be overwritten in the next line.

Reply:
you're right @hcho3 , I thought so too, but since it was part of original code and wanted the changes to be as minimally invasive I didn't. Have changed it 👍🏾